accounting/src/delta/combine.rs

```diff
@@ -66,23 +69,43 @@ pub fn combine_amount_delta(
 
 #[cfg(test)]
 pub mod test {
+    use crate::DataDeltaUndo;
+
     use super::*;
     use common::primitives::{amount::UnsignedIntType, signed_amount::SignedIntType};
 
     #[test]
     #[rustfmt::skip]
     fn test_combine_data_with_delta() {
-        let some_data_create = Some(DataDelta::Create(Box::new('b')));
-        let some_data_modify = Some(DataDelta::Modify(Box::new('b')));
+        let delta_create = Some(DeltaMapElement::Delta(DataDelta::Create('b')));
+        let delta_modify = Some(DeltaMapElement::Delta(DataDelta::Modify('a', 'b')));
+        let delta_delete= Some(DeltaMapElement::Delta(DataDelta::Delete('b')));
+
+        assert_eq!(combine_data_with_delta::<i32>(None, None),                  Ok(None));
+        assert_eq!(combine_data_with_delta(None,        delta_create.as_ref()), Ok(Some('b')));
+        assert_eq!(combine_data_with_delta(None,        delta_modify.as_ref()), Err(Error::ModifyNonexistingData));
+        assert_eq!(combine_data_with_delta(None,        delta_delete.as_ref()), Err(Error::RemoveNonexistingData));
+        assert_eq!(combine_data_with_delta(Some(&'a'),  None),                  Ok(Some('a')));
+        assert_eq!(combine_data_with_delta(Some(&'a'),  delta_create.as_ref()), Err(Error::DataCreatedMultipleTimes));
+        assert_eq!(combine_data_with_delta(Some(&'a'),  delta_modify.as_ref()), Ok(Some('b')));
+        assert_eq!(combine_data_with_delta(Some(&'a'),  delta_delete.as_ref()), Ok(None));
```

Comment:
So `Some('a')` then `Delete('b')` is `None`?

Reply:
Please include the content in the variable name so that such mistakes don't happen.